database/database.go

```diff
@@ -24,7 +24,11 @@ type Database interface {
 
 	// SaveTx will be called to save each transaction contained inside a block.
 	// An error is returned if the operation fails.
-	SaveTx(tx *types.Tx) error
+	SaveTx(tx *types.Tx, partition_id int64) error
```

Comment:
As above, I think that instead of editing a method inside the `Database` interface we should instead create a new method inside the PostgreSQL implementation class named `SaveTxInsidePartition`. This will allow to get the specific partition. For the PostgreSQL database then, the `SaveTx` method will simply: 
1. get the latest partition available 
2. call `SaveTxInsidePartition` with the latest partition